tests/lib/c_lib/src/main.c

```diff
@@ -319,6 +322,340 @@ void test_bsearch(void)
+ *
+ */
+
+void test_atoi(void)
+{
+	char *str = "123";
+	int ret;
+
+	ret = atoi(str);
+	zassert_true((ret == 123), "atoi 123");
+}
+
+/**
+ *
+ * @brief Test isalnum function
+ *
+ */
+void test_isalnum(void)
+{
+	int str1 = '1',  str2 = 'a', str3 = '!';
+
+	zassert_true(isalnum(str1), "isalnum '1'");
+	zassert_true(isalnum(str2), "isalnum 'a'");
+	zassert_true(!(isalnum(str3)), "isalnum '!'");
+}
+
+/**
+ *
+ * @brief Test isalpha function
+ *
+ */
+void test_isalpha(void)
+{
+	int ch1 = '1', ch2 = 'b';
+
+	zassert_true(!(isalpha(ch1)), "isalpha '1'");
+	zassert_true(isalpha(ch2), "isalpha 'b'");
+}
+
+/**
+ *
+ * @brief Test isdigit function
+ *
+ */
+void test_isdigit(void)
+{
+	int ch1 = '1', ch2 = 'a';
+
+	zassert_true(isdigit(ch1), "isdigit '1'");
+	zassert_true(!(isdigit(ch2)), "isdigit 'a'");
+}
+
+/**
+ *
+ * @brief Test isgraph function
+ *
+ */
+void test_isgraph(void)
+{
+	int ch = '1';
+	int sh = ' ';
+
+	zassert_true(isgraph(ch), "isgraph 1");
+	zassert_true(!(isgraph(sh)), "isgraph ' '");
+}
+
+/**
+ *
+ * @brief Test isprint function
+ *
+ */
+void test_isprint(void)
+{
+	int ch = 0x7f, sh = '1';
+
+	zassert_true(!(isprint(ch)), "isprint 0x7f");
+	zassert_true(isprint(sh), "isprint '1'");
+}
+
+/**
+ *
+ * @brief Test isspace function
+ *
+ */
+void test_isspace(void)
+{
+	int ch = 'a', sh = ' ';
+
+	zassert_true(!(isspace(ch)), "isspace 'a'");
+	zassert_true(isspace(sh), "isspace ' '");
+}
+
+/**
+ *
+ * @brief Test isupper function
+ *
+ */
+void test_isupper(void)
+{
+	int ch = 'a', sh = 'A';
+
+	zassert_true(!(isupper(ch)), "isupper 'a'");
+	zassert_true(isupper(sh), "isupper 'A'");
+}
+
+/**
+ *
+ * @brief Test isxdigit function
+ *
+ */
+void test_isxdigit(void)
+{
+	int ch1 = 'e', ch2 = 'z';
+
+	zassert_true(isxdigit(ch1), "isxdigit 'e'");
+	zassert_true(!(isxdigit(ch2)), "isxdigit 'z'");
+}
+
+/**
+ *
+ * @brief Test menchr function
+ *
+ */
+static const char str[] = "testfunction";
+void test_memchr(void)
+{
+	int ch1 = 'e', ch2 = 'z';
+	char *ret;
+
+	ret = memchr(str, ch1, strlen(str));
+	zassert_not_null(ret, "memchr 'testfunction' 'e' ");
+
+	ret = memchr(str, ch2, strlen(str));
+	zassert_is_null(ret, "memchr 'testfunction' 'z'");
+}
+
+/**
+ *
+ * @brief Test memcpy function
+ *
+ */
+void test_memcpy(void)
+{
+	char *ptr = NULL;
+	char arr[10] = "1111111111";
+	int LEN = 10;
+
+	ptr = (char *)malloc(sizeof(char) * LEN);
+	memset((int *)ptr, '0', LEN * sizeof(char));
+	memcpy(ptr, "1111111111", sizeof(char)*LEN);
+	zassert_true((memcmp(ptr, arr, sizeof(char)*LEN)) == 0,
+		     "memcpy failed");
+	free(ptr);
+	ptr = NULL;
+}
+
+/**
+ *
+ * @brief test memmove function
+ *
+ */
+void test_memmove(void)
+{
+	char str1[7] = "aabbcc";
+	char str2[7] = "aaaaaa";
+
+	memmove(str1+2, str1, 4);
+	zassert_true((memcmp(str1, str2, sizeof(char)*7)) == 0,
+		     "memmove failed");
+}
+
+/**
+ *
+ * @brief test strcat function
+ *
+ */
+void test_strcat(void)
+{
+	char str[20] = "hello";
+	char *src = "world";
```

Comment:
Not sure why we are mixing array and pointer expressions. Should stick with either one of them.

Reply:
@stephanosio thanks, I fixed those points by your comments. It's useful for me.